src/main/kotlin/com/hypto/iam/server/authProviders/AuthProviderRegistry.kt

```diff
@@ -0,0 +1,13 @@
+package com.hypto.iam.server.authProviders
+
+import com.hypto.iam.server.security.AuthenticationException
+
+object AuthProviderRegistry {
+    private val providerRegistry: Map<String, BaseAuthProvider> = buildMap {
+        put("Google", GoogleAuthProvider)
+    }
+
+    fun getProvider(providerName: String) = providerRegistry[providerName] ?: throw AuthenticationException(
+        "Invalid issuer. No provider registered providerName=[$providerName]."
+    )
```

Comment:
`AuthenticationException` is being thrown out of context. Registry does not know that the reason for `providerName` being in `providerRegistry` map is authentication failure